tests/system/utils.py

```diff
@@ -39,3 +40,121 @@ def get_resource(resource):
                 raise Exception
         except Exception:
             raise DCOSException("Can't read from resource: {0}. Please check that it exists.".format(resource))
+
+
+class FaultDomain:
+    """
+    High-level representation of a fault domain
+    """
+    def __init__(self, config):
+        # Make sure config is a dict
+        if not type(config) is dict:
+            config = {}
+
+        # Extract fault domain
+        fault_domain = config.get('fault_domain', {})
+
+        # Extract zone/region
+        self.zone = fault_domain.get('zone', {}).get('name', 'default')
+        self.region = fault_domain.get('region', {}).get('name', 'default')
+
+
+def get_cluster_local_domain():
+    """Contacts the DC/OS mesos master and returns it's faultDomain configuration (aka "local domain")
+    """
+    master = mesos.get_master()
+    return FaultDomain(master.state().get('domain', {}))
+
+
+def get_cluster_agent_domains():
+    """Returns a dictionary with the slave IDs in the cluster and their corresponding
+       fault domain information
+    """
+    slave_domains = {}
+
+    # Populate slave_domains with the ID and the corresponding domain for each slave
+    master = mesos.get_master()
+    for slave in master.slaves():
+        slave_domains[slave['id']] = FaultDomain(slave._short_state.get('domain', None))
+    return slave_domains
+
+
+def get_all_cluster_regions():
+    """Returns a dictionary with all the regions and their zones in the cluster
+    """
+    domain_regions = {}
+
+    # Populate all the domain zones and regions found in the cluster
+    for domain in get_cluster_agent_domains().values():
+        if domain.region not in domain_regions:
+            domain_regions[domain.region] = []
+        if domain.zone not in domain_regions[domain.region]:
+            domain_regions[domain.region].append(domain.zone)
+
+    # Return dictionary
+    return domain_regions
+
+
+def get_biggest_cluster_region():
+    """Returns a tuple with the name of the biggest region in the cluster and the zones in it
```

Comment:
How is bigger defined?

Reply:
Number of zones. Updated the documentation.